<<<<<<< HEAD
# role-has-required-aria-props
=======
# jsx-a11y/role-has-required-aria-props

ðŸ’¼ This rule is enabled in the following configs: â˜‘ï¸ `recommended`, ðŸ”’ `strict`.

<!-- end auto-generated rule header -->
>>>>>>> daf1795927da2d71e7fda32ee847e5794b9052d5

Elements with ARIA roles must have all required attributes for that role.

## Rule details

This rule takes no arguments.

### Succeed
```jsx
<!-- Good: the checkbox role requires the aria-checked state -->
<span role="checkbox" aria-checked="false" aria-labelledby="foo" tabindex="0"></span>
```

### Fail

```jsx
<!-- Bad: the checkbox role requires the aria-checked state -->
<span role="checkbox" aria-labelledby="foo" tabindex="0"></span>
```

## Accessibility guidelines
- [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)

### Resources
- [ARIA Spec, Roles](https://www.w3.org/TR/wai-aria/#roles)
- [Chrome Audit Rules, AX_ARIA_03](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_03)
